tests/python/contrib/test_hexagon/test_batch_flatten.py

```diff
@@ -0,0 +1,130 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import numpy as np
+import pytest
+
+import tvm
+import tvm.testing
+import tvm.topi.hexagon.slice_ops as sl
+from tvm import te, topi
+from tvm.contrib.hexagon.build import HexagonLauncher
+from tvm.topi import testing
+
+from .infrastructure import allocate_hexagon_array
+
+
+def n11c_1024c_1d(n, h, w, c):
+    return [n, h, w, c // 1024, tvm.te.AXIS_SEPARATOR, c % 1024]
+
+
+def nc_1024_1d(n, c):
+    return [n, c // 1024, tvm.te.AXIS_SEPARATOR, c % 1024]
+
+
+def transform_numpy(arr_np, layout):
+    if layout == "nhwc":
+        return arr_np
+    elif layout == "n11c-1024c-1d":
+        N, H, W, C = arr_np.shape
+        return arr_np.reshape([N, H, W, C // 1024, 1024])
+    elif layout == "nc-1d":
+        N, C = arr_np.shape
+        return arr_np.reshape([N, C // 1024, 1024])
+
+
+@tvm.testing.fixture
+def transformed_expected_output_np(expected_output_np, output_layout):
+    return transform_numpy(expected_output_np, output_layout)
+
+
+class BaseTestBatchFlatten:
```

Comment:
Separating parameters out into a base class for testing isn't required.  I use this if I want to define several groups of parameters with associated names (e.g. input sizes used by a specific model), but since there's only one subclass of `BaseTestBatchFlatten`, I'd move the contents into `TestBatchFlatten` and remove `BaseTestBatchFlatten` altogether.

There's also a lot of repetition between the test cases, so it isn't immediately clear that these are only varying the input shape and not the layouts.  I'd rearrange them so that the shape is emphasized as the parameter being varied.

```python
input_shape = tvm.testing.parameter(
    (1, 1, 1, 2048),
    (1, 2, 4, 2048),
    (1, 8, 8, 1024),
    (2, 4, 8, 1024),
    (2, 3, 5, 2048),
)
input_layout, input_axis_sep = tvm.testing.parameters(("n11c-1024c-1d", [4]))
output_layout, output_axis_sep = tvm.testing.parameters( ("nc-1d", [2]))
data_type = tvm.testing.parameter("float16")
```

Reply:
Thank you. I have updated the test cases in a new commit.